skrub/_datetime_encoder.py

```diff
@@ -325,10 +365,80 @@ def _check_params(self):
                 f"'resolution' options are {allowed}, got {self.resolution!r}."
             )
 
+        if isinstance(self.add_periodic, list):
+            if not all([_ in ["day", "year"] for _ in self.add_periodic]):
+                raise ValueError(f"Unsupported value found in {self.add_periodic}.")
+
+        if isinstance(self.add_periodic, str):
+            if self.add_periodic not in ["day", "year"]:
+                raise ValueError(
+                    f"Unsupported value {self.add_periodic} for add_periodic"
+                )
+            self.add_periodic = [self.add_periodic]
+
+        if self.add_periodic is None:
+            self.add_periodic = []
+
     def _more_tags(self):
         return {"preserves_dtype": []}
 
     def __sklearn_tags__(self):
         tags = super().__sklearn_tags__()
         tags.transformer_tags = TransformerTags(preserves_dtype=[])
         return tags
+
+
+class PeriodicEncoder(SingleColumnTransformer):
+    def __init__(self, kind: str = "spline", period=24, n_splines=None, degree=3):
```

Comment:
it seems a bit strange to have "n_splines" and "degree" parameters if the kind can be circular. maybe we should have a separate splineencoder and circular encoder classes. or just the spline one at first